### PR TITLE
.number modifier should work with select, radio, checkbox (fix #4018)

### LINKED ISSUE #4018: Model .number modifier fails to cast selected value of select element.
### Vue.js version

2.0.3
### Reproduction Link

https://jsfiddle.net/e4mxqbfv/1/
### Steps to reproduce

Change the selected option an look at the output in `<pre>` element below.
### What is Expected?

Selected number type should be a `Number` because `<select>` element is bound to a variable (model) and has `.number` modifier: `<select v-model.number="perpage">`.
### What is actually happening?

Vue fails to cast the selected value to `Number`, it remains a `String`.
### Workarounds

Some people have suggested binding the value `<option :value="10">10</option>`, but that's a workaround, not a fix. It's be great if the modifier worked on `<select>` too or at least there would be a notice in the documentation.

### PR DESCRIPTION
**Pull request by defcc**
fix #4018

### CODE REVIEW

File: src/platforms/web/compiler/directives/model.js [hunk 1]
```diff
@@ -26,19 +26,23 @@ export default function model (
...
   }
   // ensure runtime directive metadata
   return true
 }
 
-function genCheckboxModel (el: ASTElement, value: string) {
+function genCheckboxModel (
+  el: ASTElement,
+  value: string,
+  modifiers: ?Object
```
Comment (HerringtonDarkholme) [inline src/platforms/web/compiler/directives/model.js]: Is modifiers type `{[key: string]: true}` more specific?  Reference: https://github.com/vuejs/vue/blob/dev/flow/compiler.js#L46

Comment (defcc) [inline src/platforms/web/compiler/directives/model.js]: I think so. I'll update it

File: src/platforms/web/compiler/directives/model.js [hunk 2]
```diff
@@ -78,15 +87,19 @@ function genRadioModel (el: ASTElement, value: string) {
       'Declare initial values in the component\'s data option instead.'
     )
   }
-  const valueBinding = getBindingAttr(el, 'value') || 'null'
+  const number = modifiers && modifiers.number
+  let valueBinding = getBindingAttr(el, 'value') || 'null'
```
Comment (posva) [inline src/platforms/web/compiler/directives/model.js]: I think a ternary could be used here

Comment (defcc) [inline src/platforms/web/compiler/directives/model.js]: Thanks, updated :)

**Pull request merged**